pkg/apis/operator.min.io/v1/helper.go

```diff
@@ -186,10 +192,10 @@ func (mi *MinIOInstance) GetHosts() []string {
 func (mi *MinIOInstance) GetServiceHost() string {
 	if mi.Spec.Zones[0].Servers == 1 {
```

Comment:
it's guaranteed that Zones will always contains at least 1 element? otherwise we should validate `len(mi.Spec.Zones) > 0`

Reply:
there's a `ensure defaults` function that does this